R/sctour.R

```diff
@@ -0,0 +1,70 @@
+
+utils::globalVariables(
+  names = c('GEX_outfile', 'meta_outfile', 'exclusion_JSON_path', 'ptime_file'),
+  package = 'CellMembrane',
+  add = TRUE
+)
+
+#' @title Run sctour
+#'
+#' @description Runs sctour on a seurat object
+#' @param seuratObj The Seurat object containing the data to be run using sctour
+#' @param GEX_outfile The GEX filename used for output of DropletUtils::write10xCounts (.h5 extension).
+#' @param meta_outfile The filename used for containing the seuratObj metadata.
+#' @param assayName Assay whose data is to be written out with DropletUtils::write10xCounts. Should be "RNA"
+#' @param exclusionList List of genes to be excluded from variable genes in sctour. Usually a RIRA GeneSet.
+#' @param exclusion_JSON_path Filename for the file containing the gene exclusion list (.json extension)
+#' @param ptime_file Filename for the pseudotime results to be written to (.csv extension)
+#' @return A Seurat object with the pseudotime results appended. 
+#' @examples
+#' \dontrun{
+#'   CreateMergedTcrClonotypeFile(seuratObj,
+#'                                outputFile = "./tcrs.csv")
+#'   sctourSeuratObj <- Run_sctour(seuratObj = seuratObj,
+#'                              GEX_outfile = "./GEX.h5",
+#'                              meta_outfile = ./meta.csv, 
+#'                              assayName = "RNA", 
+#'                              exclusionList = RIRA::GetGeneSet('VariableGenes_Exclusion.2'),
+#'                              exclusion_JSON_path = ./exclusionList.json,
+#'                              ptime_file = NULL)
+#' }
+#' @export
+
+Run_sctour <- function(seuratObj=NULL,
```

Comment:
I would consider whether all of these inputs needs to exist, or what happens if something is NULL? For example, it doesnt seem like the user really should need to supply a ptime_file, should they? Maybe if the input is null, we should make a tempfile() and use this? The end file is read and stored in metadata anyway, isnt it?